### ISSUE TITLE
Disposing the server causes "rejected promise not handled within 1 second: Error [ERR_STREAM_WRITE_AFTER_END]"

### ISSUE THREAD

**Issue by DanTup** [microsoft/vscode-languageserver-node#723]
Disposing the LSP client seems to trigger this error. I see it in the Dart server when we restart the server (eg. if you change something that requires changing the command line arguments), but I can also reproduce it in an integration test using the source repo/test server here.

```ts
'use strict';

import * as child_process from "child_process";
import * as path from 'path';
import * as lsclient from 'vscode-languageclient/node';

suite('Client integration', () => {

	const serverModule = path.join(__dirname, './servers/testServer.js');
	const documentSelector: lsclient.DocumentSelector = [{ scheme: 'lsptests', language: 'bat' }];
	const middleware = {};
	const clientOptions: lsclient.LanguageClientOptions = {
		documentSelector,
		synchronize: {},
		initializationOptions: {},
		middleware
	};

	test.only('Can dispose and re-create language server', async () => {
		while (true) {
			const process = child_process.spawn(`node`, [serverModule, '--stdio']);
			process.stderr.on("data", (data) => console.error(data.toString()));
			process.on("exit", () => console.log(`LSP Server quit!`));

			console.log(`Creating server`);
			const client = new lsclient.LanguageClient(
				'test svr',
				'Test Language Server',
				() => Promise.resolve({ reader: process.stdout, writer: process.stdin }),
				clientOptions,
			);

			console.log(`Starting server`);
			const disp = client.start();

			console.log(`Waiting for server to be ready`);
			await client.onReady();

			console.log(`Waiting a little while before disposing...`);
			await new Promise((resolve) => setTimeout(resolve, 1000));

			console.log(`Disposing server`);
			disp.dispose();

			await new Promise((resolve) => setTimeout(resolve, 1000));
			console.log(`...`);
		}
	}).timeout(99999);
});
```

<img width="1059" alt="Screenshot 2021-01-07 at 15 22 46" src="https://user-images.githubusercontent.com/1078012/103909960-37053180-50fc-11eb-81af-6942b028d165.png">

I'm not certain, but it seems like it might be when it tries to send `{"jsonrpc":"2.0","method":"exit"}`.

**Comment by DanTup** [comment]
Commenting out this call to `exit` does stop the error occurring. Perhaps `connection.shutdown` is closing the connection so `exit` can't be written?

<img width="784" alt="Screenshot 2021-01-07 at 15 32 53" src="https://user-images.githubusercontent.com/1078012/103911190-a29bce80-50fd-11eb-8c77-cdc3c5a5b411.png">

**Comment by dbaeumer** [comment]
Yes, this got addressed but in the upcoming 3.17 release.
